symengine/mp_class.h

```diff
@@ -4,14 +4,18 @@
 #include <symengine/symengine_config.h>
 #include <symengine/mp_wrapper.h>
 
-#if SYMENGINE_INTEGER_CLASS == SYMENGINE_PIRANHA
+#ifdef HAVE_SYMENGINE_PIRANHA
 #include <piranha/mp_integer.hpp>
 #include <piranha/mp_rational.hpp>
```

Comment:
What I want to highlight is that this header is included everywhere. So, your changes will have to be moved to another file.